sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeUtils.scala

```diff
@@ -498,6 +498,11 @@ object DateTimeUtils {
     false
   }
 
+  lazy val validTimezones = TimeZone.getAvailableIDs().toSet
+  def isValidTimezone(timezoneId: String): Boolean = {
+    validTimezones.contains(timezoneId)
```

Comment:
Java does a case-sensitive check, which means Hive does too.  I don't think we want to write out a timezone w/ the wrong capitalization, and then have another tool throw an error.

```scala
scala> val tzId = "America/Los_Angeles"
tzId: String = America/Los_Angeles

scala> java.util.TimeZone.getTimeZone(tzId).getID()
res1: String = America/Los_Angeles

scala> java.util.TimeZone.getTimeZone(tzId.toLowerCase()).getID()
res2: String = GMT
```

https://github.com/apache/hive/blob/master/ql/src/java/org/apache/hadoop/hive/ql/io/parquet/timestamp/NanoTimeUtils.java#L167

We could try to auto-convert the user's timezone to the correct capitilazation, but do you think that is worth it?